### ISSUE TITLE
Events in the calendar are different from events in the source array

### ISSUE THREAD

**Issue by TimonLukas** [CroudSupport/vue-fullcalendar#97]
While implementing something using this plugin, I tried to update events by using the source array instead of relying on the fullcalendar methods. Basically, I wanted to find the element in the events array that has the same ID as the event I am trying to change, and then replace that with a new version. Problem is that the elements in the source array do not actually have all of the data in them which they theoretically do in fullcalendar itself.

Now, I am not too sure what the best way to fix this would be. It would be easy enough to do this on mount (e. g. by overwriting the array with the clientEvents array from the fullcalendar instance). But this is not an elegant solution. And I can't think of a better way to get newly inserted events to have all the data than doing this again.

So, would this be worth it to implement? It wouldn't really be backwards compatible, so it would have to be put behind a toggle, or in the next major version. I do think that it would allow your event handling to be decoupled further from fullcalendar (which is obviously a good thing), but I am not sure if it's worth the trade-offs.

**Comment by BrockReece** [comment]
Hi @TimonLukas 

Thanks for opening an issue.

Is the problem that updates to an object in the events array in the parent component is not being reflected in the calendar?

**Comment by TimonLukas** [comment]
@BrockReece Not quite. The problem is that event objects are simply different in the calendar and in the events array. This makes sense, since fullcalendar does some internal processing with events (e. g. giving them a unique ID).

But this also introduces some problems. Say you want to update an calendar event by updating it in the events array instead of using the corresponding fullcalendar method. You would search the events array for an entry with the same ID as the one you are trying to modify, and then modify that. But you can't do that since the events in the array are not updated to reflect the changes made by fullcalendar.

**Comment by BrockReece** [comment]
Does fullcalendar assign an ID if the event object already has one? If not
could you assign each event an Id before passing it through to calendar
component?

**Comment by TimonLukas** [comment]
I don't think it does. But I am not sure if it's a good idea to change the way this works.

The point I am trying to make is more that the data in the event array is inconsistent with the data in fullcalendar. This is kinda contrary to the normal two-way reactive binding, since right now it's just one-way (from the events array to fullcalendar).

**Comment by BrockReece** [comment]
Yeah I think I see the point, but we aren't claiming to offer a two way binding with the calendar component though, instead we emit events that mirror fullcalendar's api (and allow users to handle everything else they need to in the `config` prop).

Also, I am wondering how it would work practically, I can see a combination of `eventAfterAllRender` and `clientEvents` would be a starting point, but I am not sure how nicely this will play with eventSources etc...

https://codesandbox.io/s/y3101967oz

**Comment by TimonLukas** [comment]
Yeah, I understand. If I do find the time I might try to implement it myself, would you consider accepting a PR for that?

**Comment by BrockReece** [comment]
Yeah, I am open to PR's
Give me a shout if you want to bounce some more ideas around.

Cheers
Brock
